test/model-definition.test.js

```diff
@@ -334,6 +333,77 @@ describe('ModelDefinition class', function() {
     });
   });
 
+  it('should report errors if a hidden property is used in where', function(done) {
+    var Child = memory.createModel('child', {}, {hidden: ['secret']});
+    Child.create({
```

Comment:
We probably need to port these fixes to lower versions. Using callback is backward-compatible.

Reply:
Yes, we should port these fixes to lower versions which are callback based.

I'd still like us to use async/await in new code, otherwise we are introducing even more work for the upcoming refactoring of juggler.

In 3.x, we should be able to use Promises instead of callbacks, it makes the test code simpler compared to callbacks.

I am not sure about 2.x, we may be able to use callbacks there too.

Reply:
☝️ 

You haven't addressed my comment above, nor responded with any counter-argument.